main.py

```diff
@@ -191,11 +194,39 @@ def create_and_run_bot(parameters, llm_api_key):
     except Exception as e:
         raise RuntimeError(f"Error running the bot: {str(e)}")
 
+def cleanup_log_file():
```

Comment:
Instead of doing this manually can you use logaru configs where we define our logger.

an example code 
```py
from loguru import logger
import sys

# Remove default logger
logger.remove()

# Add console output
logger.add(sys.stdout, format="{time} {level} {message}")

# Add file output with rotation
logger.add(
    "app.log",
    rotation="500 MB",                # Rotate when file reaches 500MB
    retention="10 days",              # Keep logs for 10 days
    compression="zip",                # Compress rotated files
    format="{time:YYYY-MM-DD HH:mm:ss} | {level} | {message}",
    level="INFO"
)

# Example usage
def main():
    while True:
        logger.info("This is an info message")
        logger.error("This is an error message")
        logger.debug("This is a debug message")

if __name__ == "__main__":
    main()
```